eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/SubmittedVariantAccessioningService.java

```diff
@@ -59,24 +61,31 @@ public SubmittedVariantAccessioningService(SubmittedVariantMonotonicAccessioning
             throws AccessionCouldNotBeGeneratedException {
         List<AccessionWrapper<ISubmittedVariant, String, Long>> dbsnpVariants = accessioningServiceDbsnp.get(variants);
         List<ISubmittedVariant> variantsNotInDbsnp = removeFromList(variants, dbsnpVariants);
-        List<AccessionWrapper<ISubmittedVariant, String, Long>> submittedVariants = accessioningService.getOrCreate(
-                variantsNotInDbsnp);
-        return joinLists(submittedVariants, dbsnpVariants);
+        if (variantsNotInDbsnp.isEmpty()) {
+            // check this special case because mongo bulk inserts don't allow inserting empty lists
+            // (accession-commons BasicMongoDbAccessionedCustomRepositoryImpl.insert would need to change)
+            return dbsnpVariants;
+        } else {
+            List<AccessionWrapper<ISubmittedVariant, String, Long>> submittedVariants = accessioningService.getOrCreate(
+                    variantsNotInDbsnp);
+            return joinLists(submittedVariants, dbsnpVariants);
```

Comment:
Are the above changes related to EVA-1576?

Reply:
tangentially related, yes. I needed this to make the other changes. it was discussed offline not to create a ticket just for this
> -- question: if a variant with hash1 is present in dbsnp and we are asked to accession a variant that has the same hash1 but some other non-identifying fields different, should we
> 1) have 2 (similar but not identical) documents, one in SVE and the other in dbsnpSVE that have the same hash
> 2) return the dbsnp accession
> ?
> because right now we are doing option 1)  basically since the start of EVA SS issuing.
> dbsnp accessions are returned only if all the object matches exactly, including taxonomy, the flags for mismatches and validation, and the rs; which is unlikely to happen
> 
> -- how can the hash be the same if the study is different?
> -- well, ok I didn't think about that. so in practice it won't happen, but we have some PRJEBs in the dbsnp side that submitters could ask to add analysis to